modules/gapi/src/backends/cpu/gcpustereo.cpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <opencv2/gapi/stereo.hpp>
+#include <opencv2/gapi/cpu/stereo.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+
+#ifdef HAVE_OPENCV_CALIB3D
+#include <opencv2/calib3d.hpp>
+#endif // HAVE_OPENCV_CALIB3D
+
+#ifdef HAVE_OPENCV_CALIB3D
+
+/** @brief Structure for the Stereo operation setup parameters.*/
+struct GAPI_EXPORTS StereoSetup {
+    double baseline;
+    double focus;
+    cv::Ptr<cv::StereoBM> stereoBM;
+};
+
+GAPI_OCV_KERNEL_ST(GCPUStereo, cv::gapi::calib3d::GStereo, StereoSetup)
+{
+    static void setup(const cv::GMatDesc&, const cv::GMatDesc&,
+                      const cv::gapi::StereoOutputFormat,
+                      std::shared_ptr<StereoSetup> &stereoSetup,
+                      const cv::GCompileArgs &compileArgs) {
+        auto stereoInit = cv::gapi::getCompileArg<cv::gapi::calib3d::cpu::StereoInitParam>(compileArgs)
+            .value_or(cv::gapi::calib3d::cpu::StereoInitParam{});
+
+        StereoSetup ss{stereoInit.baseline,
+                       stereoInit.focus,
+                       cv::StereoBM::create(stereoInit.numDisparities,
+                       stereoInit.blockSize)};
+        stereoSetup = std::make_shared<StereoSetup>(ss);
+    }
+    static void run(const cv::Mat& left,
+                    const cv::Mat& right,
+                    const cv::gapi::StereoOutputFormat oF,
+                    cv::Mat& out_disp,
+                    StereoSetup stereoSetup) {
+        cv::Mat disp;
+        stereoSetup.stereoBM->compute(left, right, disp);
+
+        switch(oF){
+            case cv::gapi::StereoOutputFormat::DEPTH_FLOAT16:
+                disp.convertTo(disp, CV_32FC1);
```

Comment:
I am wondering does it work? How do the OpenCV know how much bits are used for fractional part?

Reply:
Fixed